src/device.jl

```diff
@@ -18,9 +25,11 @@ function device_reset(dev::Integer)
     for p in todelete
         delete!(cuda_ptrs, p)
     end
-    # Reset the device
-    device(dev)
-    rt.cudaDeviceReset()
+
+    # no need to reset the entire device, just reset the primary context
+    pctx = CuPrimaryContext(dev)
+    CUDAdrv.reset(pctx)
```

Comment:
I think the release after the reset is fine. See https://github.com/JuliaGPU/CUDAdrv.jl/pull/19#issuecomment-282956880 for what I think is happening here.